R/roclet-doxygen.R

```diff
@@ -0,0 +1,138 @@
+#' @title Replace a value for a given tag on file in memory
+#' @description Scan the lines and change the value for the named tag if one line has this tag, 
+#'    add a line at the end if no line has this tag and return a warning if several lines
+#'    matching the tag
+#' @param fileStrings A vector with each string containing a line of the file
+#' @param tag The tag to be searched for 
+#' @param newVal The new value for the tag
+#' @return The vector of strings with the new value
+#' @examples
+#' fakeFileStrings <- c("Hello = world","SURE\t= indeed","Hello = you")
+#' 
+#' expect <- warning(ReplaceTag(fakeFileStrings,"Hello","me"))
+#' 
+#' newFake <- ReplaceTag(fakeFileStrings,"SURE","me")
+#' expect <- equal(length(newFake), length(fakeFileStrings))
+#' expect <- equal(length(grep("SURE",newFake)), 1)
+#' expect <- equal(length(grep("me",newFake)), 1)
+#' 
+#' newFake <- ReplaceTag(fakeFileStrings,"Bouh","frightened?")
+#' expect <- equal(length(newFake), length(fakeFileStrings)+1)
+#' expect <- equal(length(grep("Bouh",newFake)), 1)
+#' expect <- equal(length(grep("frightened?",newFake)), 1)
+#' @keywords internal 
+ReplaceTag <- function(fileStrings,tag,newVal){
+    iLine <- grep(paste0("^",tag,"\\>"),fileStrings)
+    nLines <- length(iLine)
+    if(nLines == 0){
+        line <- paste0(tag,"\t= ",newVal)
+        iLine <- length(fileStrings)+1
+    }else if (nLines > 0){
+        line <- gsub("=.*",paste0("= ",newVal),fileStrings[iLine])
+        if(nLines >1){
+            warning(paste0("File has",nLines,"for key",tag,"check it up manually"))
+        }
+    }
+    fileStrings[iLine] <- line
+    return(fileStrings)
+}
+#' Prepares the R package structure for use with doxygen
+#' @description Makes a configuration file in inst/doxygen
+#'     and set a few options: 
+#'     \itemize{
+#'        \item{EXTRACT <- ALL = YES}
+#'        \item{INPUT = src/}
+#'        \item{OUTPUT <- DIRECTORY = inst/doxygen/}
+#'     }
+#' @param rootFolder The root of the R package
+#' @note Please note this implies the installation of doxygen on the system, 
+#'       accessible on the command line by `doxygen`
+#' @return NULL
+#' @examples 
+#' \dontrun{
+#' doxygen_init()
+#' }
+#' @keywords internal
```

Comment:
That's what all the `use_xyz()` functions do - it's short hand for use "xyz" with this package.